pubsub.go

```diff
@@ -183,6 +190,7 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 		topics:           make(map[string]map[peer.ID]struct{}),
 		peers:            make(map[peer.ID]chan *RPC),
 		topicVals:        make(map[string]*topicVal),
+		validateQ:        make(chan *validateReq, 32),
```

Comment:
Should the size of `validateQ` the same as or greater than `defaultValidateThrottle`? Though it seems `defaultValidateThrottle` is the potential max number of the concurrent goroutines.

Reply:
I don't think so, or we will have the same problems with overloading we have previously.

Reply:
So to explain: the throttle controls the number of active asynchronous validation requests, which can take a while to complete.
The queue controls the front end to the validation pipeline, which can  drop messages if we are overloaded - see #172.

Reply:
Big thanks for the clear explanation. Are my following understanding correct?
- `validatorQ` with the workers serve as the rate-limiting gate
- Except for the removal of the timeout, the asynchronous verification part is roughly the same as the one before this PR(except for the fact that signature verification is done synchronously by the workers).

Reply:
yes, this is correct.